.gitignore

```diff
@@ -73,9 +76,15 @@ Session.vim
 tags
 ### VisualStudioCode ###
 .vscode/*
-.history
-# End of https://www.gitignore.io/api/go,vim,emacs,visualstudiocode
-pulumi-kubernetes-operator
-### IntelliJ
+!.vscode/settings.json
```

Comment:
Knowing nothing about VSCode configuration, why do we exclude `.vscode/*` and then add back certain files. Do we expect that contributors will have other non-checked in files in `.vscode`?

Reply:
Agreed. Perhaps we can just remove the .vscode ignore?

Reply:
Looking around on the Web, the approach shown here is the norm (see links in description). Assumedly there are good reasons, can we just trust the crowd here?

Reply:
These are all distinct file names in all my .vscode folders, via a `find | sort | uniq` command.

```
configurationCache.log
cspell-azd-dictionary.txt
cspell-dictionary.txt
cspell-github-user-aliases.txt
cspell-notice.txt
cspell-templates.txt
cspell.global.yaml
cspell.yaml
dryrun.log
extensions.json
launch.json
settings.json
targets.log
tasks.json
```

I don't even know what `cspell` is, and we don't want the `log` files. So the approach here makes sense, based on my n=1 sample. I guess we could also allow all .json and .yaml files?

Reply:
I would advocate for us to stick to the well-known approach in our first iteration, which is to whitelist a few files but ignore everything else. Folks may raise a follow-up PR as we learn about other stuff.